### ISSUE TITLE
Dropdown misplacement

### ISSUE THREAD

**Issue by tamis-laan** [qtile/qtile#1920]
I have the following scratchpad configuration:

```python
scratchpad = ScratchPad("scratchpad", [
    DropDown("jit", "qutebrowser -T -s statusbar.show never -s tabs.show never meet.jit.si",
                x=0.7, y=0.0, width=0.29, height=0.3, on_focus_lost_hide=False)
])
keys = [
    Key([mod1], 'v', lazy.group['scratchpad'].dropdown_toggle('jit')),
]
```
When pressing `mod+v` the dropdown is misplaced on the left side of the screen instead of the right side. The `x` `y` `width` `height` parameters have no impact on the dropdown position and size. When moving the window it does jump to the right position.

**Comment by tamis-laan** [comment]
For reference: 
```
> qtile --version
0.16.2.dev43+g2f5da5ca
```

**Comment by m-col** [comment]
Just so we can more easily test this, do you see the same behaviour if you use a plain xterm instead of qutebrowser?

**Comment by tamis-laan** [comment]
My other drop downs work and they are terminal based, I use the termite terminal.  However both `qutebrowser` and `jitsi-meet-desktop` which is an electron application have the above stated problem.

**Comment by m-col** [comment]
Right. I tested it with qutebrowser and what I notice is that if I resize and reposition qutebrowser and then close it with Ctrl-Q, the old position and size is retained when it is opened up again. qutebrowser is remembering the user-specified geometry and requesting it to be restored. Normally Qtile respects this for floating windows, so it looks like we are respecting these requests for windows in the scratchpad too, when instead we should be forcing a geometry onto them. Shouldn't be too hard to track this down.

**Comment by tamis-laan** [comment]
I would definitely expect forced geometry in case of dropdowns. This will also make it less likely for the user to accidentally move or change the geometry of the dropdown when using a mouse.

**Comment by m-col** [comment]
Does it get the correct geometry after you have hidden the window and shown it again, after it has been spawned? i.e. is just the very first time it appears where the geometry is wrong?

**Comment by tamis-laan** [comment]
Nope, it keeps its geometry and location. However when I move the window it jumps to the geometry specified in the config.

**Comment by tamis-laan** [comment]
Although I'm not familiar with the `qtile` code base, looking at `scratchpad.py` I can see that when the dropdown is toggled the geometry is set every time before the window is unhidden: 
https://github.com/qtile/qtile/blob/2f5da5ca9de473b046959e6af7495e3193651d21/libqtile/scratchpad.py#L181
```python
    def show(self):
        """
        Like WindowVisibilityToggler.show, but before showing the window,
        its floating x, y, width and height is set.
        """
        if (not self.visible) or (not self.shown):
            win = self.window
            screen = win.qtile.current_screen
            # calculate windows floating position and width/height
            # these may differ for screens, and thus always recalculated.
            win.x = int(screen.dx + self.x * screen.dwidth)
            win.y = int(screen.dy + self.y * screen.dheight)
            win.float_x = win.x
            win.float_y = win.y
            win.width = int(screen.dwidth * self.width)
            win.height = int(screen.dheight * self.height)

            # SHOW
            WindowVisibilityToggler.show(self)
```
However the `qutebrowser` window does not spawn in the right location and when moving the window it does "jump" back into the correct location. This makes me think that `qtile` is tracking the geometry/position correctly but `qutebrowser` is also tracking it's geometry/location and overriding it as soon as it is hidden/shown.

By swapping the order it might give at least partial correct results:
```python 
    def show(self):
        """
        Like WindowVisibilityToggler.show, but before showing the window,
        its floating x, y, width and height is set.
        """
        if (not self.visible) or (not self.shown):
            # SHOW
            WindowVisibilityToggler.show(self)

           # SET GEOMETRY
            win = self.window
            screen = win.qtile.current_screen
            # calculate windows floating position and width/height
            # these may differ for screens, and thus always recalculated.
            win.x = int(screen.dx + self.x * screen.dwidth)
            win.y = int(screen.dy + self.y * screen.dheight)
            win.float_x = win.x
            win.float_y = win.y
            win.width = int(screen.dwidth * self.width)
            win.height = int(screen.dheight * self.height)
```

**Comment by ramnes** [comment]
Fixed in #1926
